<a href="https://colab.research.google.com/github/MariaArangoOchoa/Anal-tica-de-Negocios-/blob/main/Reto2_Mar%C3%ADaArango.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Reto 2**
0. Cargar las librerias de referencia para el modelo

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.tree import DecisionTreeClassifier

from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


1.Se procede con la carga de los datos del archivo


In [ ]:
nxl='/content/drive/MyDrive/Colab Notebooks/Base de Datos .xlsx'
XDB=pd.read_excel(nxl,sheet_name=0)
XDB=XDB.dropna()
XDB.head(100)

nxl='/content/drive/MyDrive/Colab Notebooks/BD1-GymData_Int.xlsx'
XDB=pd.read_excel(nxl,sheet_name=0)
XDB=XDB.dropna()
XDB.head(100)

XDB=XDB[['Age', 'Income', 'Hours', 'Enroll']]
sns.pairplot(XDB,hue='Enroll', size=3, vars=['Age', 'Income', 'Hours'])



[[1.00000e+00 5.47243e+03 4.91500e+04 8.27400e+02]
 [0.00000e+00 9.13073e+03 4.78060e+04 8.63550e+02]
 [0.00000e+00 4.45067e+03 4.60500e+04 1.18005e+03]
 ...
 [0.00000e+00 3.58586e+03 4.22500e+04 2.25195e+03]
 [1.00000e+00 1.83395e+03 6.02480e+04 2.00025e+03]
 [0.00000e+00 6.20597e+03 5.85200e+04 3.66750e+03]]
[[1]
 [1]
 [1]
 [1]
 [1]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [1]
 [1]
 [1]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [1]
 [0]
 [1]
 [1]
 [0]
 [1]
 [0]
 [1]
 [1]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [1]
 [0]
 [1]
 [0]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0

2. Creación del modelo - Decision Tree

In [ ]:
import matplotlib.pyplot as plt
from sklearn.tree import DecisionTreeClassifier
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()
XD[:, 0] = le.fit_transform(XD[:, 0])

mar=DecisionTreeClassifier(criterion='gini',max_depth=4)
mar.fit(XD,yd)
ydp=mar.predict(XD)


from sklearn.metrics import confusion_matrix
cm=confusion_matrix(yd,ydp)
print(cm)

VN=cm[0,0]
FP=cm[0,1]
FN=cm[1,0]
VP=cm[1,1]

Exactitud=(VP+VN)/(VN+VP+FN+FP)
print("La Exactitud Alcanzada por el Modelo es: ",Exactitud)
Sensibilidad=VP/(FN+VP)
print("La Sensibilidad Alcanzada por el Modelo es: ",Sensibilidad)
Especificidad=VN/(VN+FP)
print("La Especificidad Alcanzada por el Modelo es: ",Especificidad)

[[334   9]
 [ 42 115]]
La Exactitud Alcanzada por el Modelo es:  0.898
La Sensibilidad Alcanzada por el Modelo es:  0.732484076433121
La Especificidad Alcanzada por el Modelo es:  0.9737609329446064


3. ¿Y dónde está el árbol?

In [ ]:
from sklearn.tree import export_graphviz
from pydotplus import graph_from_dot_data

vs=['College', 'FoodSpend', 'Income', 'TravelSpend']
dot_graph=export_graphviz(mar,feature_names=vs)
graph=graph_from_dot_data(dot_graph)
graph.write_png('arbol.png')

True

4. Se procede con el pronóstico de los nuevos clientes


In [ ]:
XDB2=pd.read_excel(nxl,sheet_name=1)
XDB2=XDB2.dropna()
XDB2.head(100)

XD2=np.array(XDB2[['College', 'FoodSpend', 'Income', 'TravelSpend']])
print(XD2)

ydp2=mar.predict(XD2)
print(ydp2)



[0 0]


**Análisis de Resultados**
De acuerdo con la base de datos suministrada y el nodo base del árbol de decisión, se logró pronosticar una exactitud aproximada al 90%, junto a una Sensibilidad que alcanzó el 73%, además de la  Especificidad, indicador que resultó con un porcentaje del 97%. Respecto a la probabilidad de que el primer cliente tenga planes de viajar el próximo año es de un 100%, mientras que la probabilidad de que el segundo cliente tenga planes de viajar el próximo año es de un 78% aproximadamente.

Más aún, las reglas del negocio que pertenece el primer cliente es TravelSpend>= 1567.875 AND Income>= 59098.0 [108]

Por otro lado, el segundo cliente es parte de las reglas del negocio de TravelSpend>= 1567.875 AND Income<= 59098.0 AND FoodSpend>= 2711.895 AND TravelSpend<= 3008.25 [136]

Se destaca tres nodos puros los cuales son
1. TravelSpend<= 1567.875 AND TravelSpend<= 1188.0 AND FoodSpend <= 4990.73 AND FoodSpend >= 4972.55 [1]
2. TravelSpend<= 1567.875 AND TravelSpend<= 1188.0 AND FoodSpend >= 4990.73 [47]
3. TravelSpend>= 1567.875 AND Income>= 59098.0 [108]